File: src/main/java/com/dh/ondot/schedule/domain/converter/RepeatDaysConverter.java
package com.dh.ondot.schedule.domain.converter;

import jakarta.persistence.AttributeConverter;
import jakarta.persistence.Converter;

import java.util.Arrays;
import java.util.SortedSet;
import java.util.TreeSet;
import java.util.stream.Collectors;

@Converter
public class RepeatDaysConverter implements AttributeConverter<SortedSet<Integer>, String> {
    @Override
    public String convertToDatabaseColumn(SortedSet<Integer> attribute) {
        if (attribute == null || attribute.isEmpty()) return "";
        return attribute.stream()
                .map(String::valueOf)
                .collect(Collectors.joining(","));
    }

    @Override
    public SortedSet<Integer> convertToEntityAttribute(String dbData) {
        if (dbData == null || dbData.isBlank()) return new TreeSet<>();
        return Arrays.stream(dbData.split(","))
                .map(String::trim)
                .map(Integer::valueOf)
                .collect(Collectors.toCollection(TreeSet::new));
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/converter/RepeatDaysConverter.kt
package com.dh.ondot.schedule.domain.converter

import jakarta.persistence.AttributeConverter
import jakarta.persistence.Converter
import java.util.SortedSet
import java.util.TreeSet

@Converter
class RepeatDaysConverter : AttributeConverter<SortedSet<Int>, String> {

    override fun convertToDatabaseColumn(attribute: SortedSet<Int>?): String {
        if (attribute.isNullOrEmpty()) return ""
        return attribute.joinToString(",")
    }

    override fun convertToEntityAttribute(dbData: String?): SortedSet<Int> {
        if (dbData.isNullOrBlank()) return TreeSet()
        return dbData.split(",")
            .map { it.trim().toInt() }
            .toCollection(TreeSet())
    }
}
